documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,83 @@
+Both mechanisms allow Kafka clients to establish authenticated sessions with Kafka brokers. 
+The authentication flow between clients, the authorization server, and Kafka brokers is different for each mechanism.
 
-A Kafka client initiates a session with the Kafka broker using the _SASL OAUTHBEARER_ mechanism for credentials exchange, where credentials take the form of an access token.
+Configure clients to use OAUTHBEARER for authentication with Kafka brokers whenever possible. 
+This mechanism provides a higher level of security because client credentials are _never_ shared with Kafka brokers. 
+Consider using PLAIN only with Kafka clients that do not support OAUTHBEARER. 
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+OAUTHBEARER and PLAIN can be enabled at the same time, on the same `oauth` listener, if both types of OAuth clients need to authenticate.
+
+.OAUTHBEARER overview
+
+Support for OAUTHBEARER is built in to many Kafka clients. 
+However, the developer must often write custom callbacks that provide access tokens for authentication, as needed.
+Strimzi provides these callbacks for standard Java Kafka clients (clients that use the Kafka producer and consumer APIs). 
+Therefore, developers do not need to provide custom callbacks in their application code. 
+
+With OAUTHBEARER, the client initiates a session with the Kafka broker for credentials exchange, where credentials take the form of a bearer token provided by the Strimzi OAuth callback code. 
+Using the callbacks, you can configure token provision in one of three ways:
+
+* Client ID and Secret (by using the OAuth 2.0 client credentials mechanism)
+
+* A long-lived access token, obtained manually at configuration time
+
+* A long-lived refresh token, obtained manually at configuration time
+
+OAUTHBEARER is automatically enabled in the `oauth` listener configuration for the Kafka broker. 
+You can set the `enableOauthBearer` property to `true`, though this is not required.
+
+[source,yaml,subs="attributes+"]
+----
+  # ...
+  authentication:
+    type: oauth
+    # ...
+    enableOauthBearer: true
+----
+
+.PLAIN overview
+
+PLAIN is a simple authentication mechanism supported by all Kafka client tools (including developer tools such as kafkacat). 
+To enable PLAIN to be used together with OAuth authentication, Strimzi includes additional server-side callbacks and calls this _OAuth 2.0 over PLAIN_. 
+
+With PLAIN, Client credentials are handled centrally, behind an authorization server that complies with OAuth 2.0 and OpenID Connect (OIDC).
+
+When used with the provided callbacks, PLAIN allows Kafka clients to authenticate with Kafka brokers using either of the following methods:
+
+* Client ID and Secret (by using the OAuth 2.0 client credentials mechanism)
+
+* A long-lived access token, obtained manually at configuration time
+
+The client must enable PLAIN and set the `username` and `password`. 
```

Comment:
```suggestion
The client must be enabled to use PLAIN authentication with a `username` and `password`. 
```

Reply:
Do we provide example clients configured for PLAIN and OAUTHBEARER?

Reply:
@mstruk - Do we provide this?

Reply:
You mean example of client configuration for using PLAIN to connect to the Kafka broker?
There's documentation for it [here](https://github.com/strimzi/strimzi-kafka-oauth/blob/main/README.md#configuring-the-kafka-client-with-sasl_plain). PLAIN mechanism doesn't require any additional Strimzi libraries on the client.

OAUTHBEARER on the other hand requires strimzi-kafka-oauth client libraries as explained [here](https://github.com/strimzi/strimzi-kafka-oauth/blob/main/README.md#configuring-the-kafka-client-with-sasl_oauthbearer)